presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -181,7 +182,7 @@ public GroupedAccumulator createGroupedAccumulator(UpdateMemory updateMemory)
         ImmutableList.Builder<Integer> aggregateInputChannels = ImmutableList.builder();
         aggregateInputChannels.addAll(inputChannels);
         maskChannel.ifPresent(aggregateInputChannels::add);
-        return new SpillableFinalOnlyGroupedAccumulator(sourceTypes, aggregateInputChannels.build(), (FinalOnlyGroupedAccumulator) accumulator);
+        return new SpillableFinalOnlyGroupedAccumulator(sourceTypes, aggregateInputChannels.build(), (FinalOnlyGroupedAccumulator) accumulator, isSpillableAccumulatorPreprocessingEnabled(session));
```

Comment:
Let's checkArg session is not null. It's interesting that session could be null when `distinct` is false by reading the context.